core/src/main/java/org/infinispan/eviction/impl/ActivationManagerImpl.java

```diff
@@ -88,7 +87,7 @@ public void onRemove(Object key, boolean newEntry) {
          } else {
             //the entry already exists in data container. It may be put during the load by the CacheLoaderInterceptor
             //so it was already activate in the private stores.
-            if (primaryOwner && persistenceManager.deleteFromAllStores(key, SHARED) && statisticsEnabled) {
+            if (primaryOwner && persistenceManager.deleteFromAllStores(key, BOTH) && statisticsEnabled) {
```

Comment:
We should really stop trying to make passivation work with shared stores: https://issues.jboss.org/browse/ISPN-7168